lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -2587,22 +2594,282 @@ spec = describe "SHARED_TRANSACTIONS" $ do
+            (Link.listAddresses @'Shared wDest) Default Empty
+        expectResponseCode HTTP.status200 rAddr
+        let addrs = getFromResponse Prelude.id rAddr
+
+        let addr0 = (head addrs) ^. #id
+        let linkList0 = Link.listTransactions' @'Shared wDest
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                (Just (apiAddress addr0))
+        rl0 <- request @([ApiTransaction n]) ctx linkList0 Default Empty
+        verify rl0 [expectListSize 2]
+        let txs0 = getFromResponse Prelude.id rl0
+        let amts0 = fmap (view #amount) txs0
+        Set.fromList amts0 `shouldBe` Set.fromList (Quantity <$> [a1, a2])
+
+        let addr1 = (addrs !! 1) ^. #id
+        let linkList1 = Link.listTransactions' @'Shared wDest
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                (Just (apiAddress addr1))
+        rl1 <- request @([ApiTransaction n]) ctx linkList1 Default Empty
+        verify rl1 [expectListSize 1]
+        let txs1 = getFromResponse Prelude.id rl1
+        let amts1 = fmap (view #amount) txs1
+        amts1 `shouldBe` (Quantity <$> [a3])
+
+        let addr2 = (addrs !! 2) ^. #id
+        let linkList2 = Link.listTransactions' @'Shared wDest
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+                (Just (apiAddress addr2))
+        rl2 <- request @([ApiTransaction n]) ctx linkList2 Default Empty
+        verify rl2 [expectListSize 3]
+        let txs2 = getFromResponse Prelude.id rl2
+        let amts2 = fmap (view #amount) txs2
+        Set.fromList amts2 `shouldBe` Set.fromList (Quantity <$> [a1, a4, a5])
+
+    it "SHARED_TRANSACTIONS_LIST_06 - filter address input side" $ \ctx -> runResourceT $ do
+        let minUTxOValue' = minUTxOValue (_mainEra ctx)
+        let a1 = minUTxOValue'
+        let a2 = fromIntegral $ oneAda * 5_000
+        let a3 = fromIntegral $ oneAda * 10_000
+        (wSrc, ApiSharedWallet (Right wDest)) <-
+             (,) <$> fixtureSharedWalletWith ctx (fromIntegral $ oneAda * 50_000)
+                 <*> emptySharedWallet ctx
+
+        -- initially empty wallet wDest should have five txs on address 0 with
+        -- just 5*a1 on it, one tx on address 1 with a2 and three txs on address
+        -- 2 with 3*a3
+        sendAmtToAddr ctx wSrc wDest a1 0
+        sendAmtToAddr ctx wSrc wDest a1 0
+        sendAmtToAddr ctx wSrc wDest a1 0
+        sendAmtToAddr ctx wSrc wDest a1 0
+        sendAmtToAddr ctx wSrc wDest a1 0
+
+        sendAmtToAddr ctx wSrc wDest a3 2
+        sendAmtToAddr ctx wSrc wDest a3 2
+        sendAmtToAddr ctx wSrc wDest a3 2
+
+        eventually "There are exactly 8 transactions for wDest" $ do
+            let linkList = Link.listTransactions' @'Shared wDest
```

Comment:
`listTransactionsFilteredByAddress` as above.

It may be worth putting this function `listTransactionsFilteredByAddress` into the `where` clause.

Reply:
`listTransactionsFilteredByAddress` now in `where` for shelley, shared and byron styles